### PR TITLE
Fix industrial miner mining up sf blocks

### LINKED ISSUE #3511: Industrial Miner mines mob simulation chamber from infinity expansion.
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

Linking the discord discussion here. I was able to confirm this is a bug.
https://discord.com/channels/565557184348422174/565569196218384385/958476080820420659

when a mob simulation chamber is within the range of the industrial miner, it mines it as gilded blackstone. the slimefun block data is still there. when you place any block back where the mob sim chamber was, it drops the chamber. 

### 📑 Reproduction Steps

1. Setup the industrial miner
2. place a mob simulation chamber below the miner within its mining range
3. start up the miner
4. it will collect it as gilded blackstone and leave the slimefun data.

### 💡 Expected Behavior

I would expect it to not mine the mob simulation chamber.

### 📷 Screenshots / Videos

_No response_

### 📜 Server Log

no errors in log

### 📂 `/error-reports/` folder

_No response_

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.18.x

### ⭐ Slimefun version

1008

### 🧭 Other plugins

_No response_

### PR DESCRIPTION
**Pull request by FN-FAL113**
## Description
Prevent the industrial miner from mining any sf blocks that uses ores as their material

## Proposed changes
- Added a short circuit condition to prevent unintentional mining of sf blocks

## Related Issues (if applicable)
- Resolves #3511 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/MiningTask.java [hunk 1]
```diff
@@ -198,7 +199,7 @@ public void run() {
                         return;
                     }
 
-                    if (miner.canMine(b.getType()) && push(miner.getOutcome(b.getType()))) {
+                    if (!BlockStorage.hasBlockInfo(b) && miner.canMine(b.getType()) && push(miner.getOutcome(b.getType()))) {
```
[the remark this answers is not in the document]
Comment (FN-FAL113) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/MiningTask.java:202]: I shall move it when I get into my pc, It seems like i I shouls have put it in there last time I checked it my logic wasn't able to reach it and focused on the task conditions

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/IndustrialMiner.java [hunk 2]
```diff
@@ -202,22 +203,22 @@ public void onInteract(Player p, Block b) {
...
-            return canMineAncientDebris.getValue();
-        } else if (version.isAtLeast(MinecraftVersion.MINECRAFT_1_17) && SlimefunTag.DEEPSLATE_ORES.isTagged(type)) {
-            return canMineDeepslateOres.getValue();
+        if (version.isAtLeast(MinecraftVersion.MINECRAFT_1_16) && block.getType() == Material.ANCIENT_DEBRIS) {
+            return canMineAncientDebris.getValue() && BlockStorage.check(block) == null;
+        } else if (version.isAtLeast(MinecraftVersion.MINECRAFT_1_17) && SlimefunTag.DEEPSLATE_ORES.isTagged(block.getType())) {
+            return canMineDeepslateOres.getValue() && BlockStorage.check(block) == null;
         } else {
-            return SlimefunTag.INDUSTRIAL_MINER_ORES.isTagged(type);
+            return SlimefunTag.INDUSTRIAL_MINER_ORES.isTagged(block.getType()) && BlockStorage.check(block) == null;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/IndustrialMiner.java:221]: You can use `BlockStorage#hasBlockInfo()` instead here.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by FN-FAL113** [comment]
Bump

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/IndustrialMiner.java [hunk 3]
```diff
@@ -202,22 +203,22 @@ public void onInteract(Player p, Block b) {
...
-            return canMineAncientDebris.getValue();
-        } else if (version.isAtLeast(MinecraftVersion.MINECRAFT_1_17) && SlimefunTag.DEEPSLATE_ORES.isTagged(type)) {
-            return canMineDeepslateOres.getValue();
+        if (version.isAtLeast(MinecraftVersion.MINECRAFT_1_16) && block.getType() == Material.ANCIENT_DEBRIS) {
+            return canMineAncientDebris.getValue() && BlockStorage.hasBlockInfo(block);
+        } else if (version.isAtLeast(MinecraftVersion.MINECRAFT_1_17) && SlimefunTag.DEEPSLATE_ORES.isTagged(block.getType())) {
+            return canMineDeepslateOres.getValue() && BlockStorage.hasBlockInfo(block);
         } else {
-            return SlimefunTag.INDUSTRIAL_MINER_ORES.isTagged(type);
+            return SlimefunTag.INDUSTRIAL_MINER_ORES.isTagged(block.getType()) && BlockStorage.hasBlockInfo(block);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/IndustrialMiner.java:221]: Shouldn't these be inverted?
The block should only be mineable if it has **NO** blockinfo.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by FN-FAL113** [comment]
woah didn't know this pr is 2 months old now

**Comment by FN-FAL113** [comment]
Re-opening this, since there are no guys who are willing to make a pr for the issue. This might be in need of review, my little time can still manage this after dealing with some stuffs irl

**Comment by FN-FAL113** [comment]
if in need of change, Will have to do another pr with a new branch since this pr's fork is removed

**Comment by Phoenix-Starlight** [comment]
Can't you restore the branch?

**Pull request closed without merging**